pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go

```diff
@@ -439,7 +439,7 @@ func (s *streamIngestionResumer) protectDestinationTenant(
 	execCfg := execCtx.(sql.JobExecContext).ExecCfg()
 	target := ptpb.MakeTenantsTarget([]roachpb.TenantID{oldDetails.DestinationTenantID})
 	ptsID := uuid.MakeV4()
-	now := execCfg.Clock.Now()
+	now := oldDetails.ReplicationStartTime
```

Comment:
The ReplicationStartTime is picked on the source cluster. If this timestamp is far enough in the past that it falls below the destination clusters' GCThreshold this protection will be considered invalid. Note, the drift between the clusters would have to be > than the GCTTL of the clusters making this unlikely.

I wonder if it is more correct to leave this as it is, but instead, wait for the replicated time to be above the protected time before performing a cutover to LATEST?

Reply:
Another related thought, we should probably leave replicated time in `SHOW TENANT WITH REPLICATION STATUS` set to DNULL until it is past the protected timestamp. Only then is it really safe to cutover the replication stream?

Reply:
> I wonder if it is more correct to leave this as it is, but instead, wait for the replicated time to be above the protected time before performing a cutover to LATEST?

ah! this is what I implemented initially :) BUT, I don't like it because it means that cutting over to `LATEST` would cutover to a time in the future, meaning, say the frontier is at T0 and the pts is at T1 (greater than T0) then we will not cutover until we see T1 which may never happen.

I know it's only on a new replication.. and the time diff should never be significant but I think that using the source cluster time is the right thing anyway: we want to protect timestamp X where X is the time of the data, not the time that the data was inserted to the dest cluster.

> The ReplicationStartTime is picked on the source cluster. If this timestamp is far enough in the past that it falls below the destination clusters' GCThreshold this protection will be considered invalid. Note, the drift between the clusters would have to be > than the GCTTL of the clusters making this unlikely.

right, clusters should not drift that much.

> Another related thought, we should probably leave replicated time in SHOW TENANT WITH REPLICATION STATUS set to DNULL until it is past the protected timestamp. Only then is it really safe to cutover the replication stream?

no need with the current state of this commit, right?

Reply:
> we want to protect timestamp X where X is the time of the data, not the time that the data was inserted to the dest cluster

I think this exposes us to a correctness issue though. Semantically do you agree that it is risky to write a PTS record at a timestamp that is not picked on the cluster it is being applied to? If we write a PTS record at a timestamp below our GCThreshold it will silently fail to protect any data and result in data loss. I think it is better to reject a cutover if our `replicated_time` is less than `retained_time`. I agree that it could be a timestamp in the future but the difference between the initial scan timestamp and `retained_time` is expected to be very very small, and if we can't replicate up to `retained_time` something is wrong. In such a situation is it worth allowing a user to cutover to a timestamp that *may* have been GC'ed?

Reply:
Spoke to @lidorcarmel offline and I stand corrected. I forgot we used the StreamSSTBatcher that does not ingest keys at the AddSST requests batch timestamp, but at the timestamp that they existed on the source cluster. Thus, it is okay to protect at a timestamp picked on the source cluster. We are still in trouble if this timestamp is below the GCThreshold of the destination cluster but that is a problem for another day.